BlueDBOnDisk/src/main/java/org/bluedb/disk/serialization/ThreadLocalFstSerializer.java

```diff
@@ -68,7 +71,7 @@ protected DefaultCoder initialValue() {
 			try {
 				byte[] serializedBytes = get().toByteArray(o);
 				validateBytesAfterSerialization(serializedBytes);
-				return serializedBytes;
+				return encryptionService.encryptOrReturn(serializedBytes);

```

Comment:
This isn't going to work. We need to pass the encryption version key to use in the method here so. Otherwise we'll potentially use different encryption mechanisms for objects that are in the same file. Same goes for the `toObject` method. As I mentioned in my original writeup to you, I think that `ReadFileManager`, `ReadWriteFileManager` , `BlueObjectInput`, and `BlueObjectOutput` are going to need to get the encryption version key once per file and then pass it into the serializer methods.

Reply:
Yeah, this isn't done, still working on the implementation details.